### PR TITLE
docs: update documentation for edgeXBuildGoApp and edgeXBuildGoParallel

### PR DESCRIPTION
**Pull request by soda480**
Documented parameters and added complete example for `edgeBuildGoApp` and `edgeXBuildGoParallel`

The HTML is **NOT** included in the repository.  The build generates the documentation; it is generated from comments in the code, comments are read and translated into markdown, then markdown is published as HTML. This only happens in the build when PRs are merged into master. 

In order to view the HTML as part of this PR, you will need to manually generate the documentation using the following commands:

```
docker container run \
--rm \
-t \
-e GRADLE_OPTS="-Dhttp.proxyHost=proxy-chain.intel.com -Dhttp.proxyPort=911 -Dhttps.proxyHost=proxy-chain.intel.com -Dhttps.proxyPort=911" \
-v $PWD:/code \
-v $HOME/.gradle:/home/gradle/.gradle \
-w /code \
gradle:6.5 \
gradle generateDocumentation
```

```
docker image build \
--build-arg http_proxy \
--build-arg https_proxy \
-t mkdocs:latest \
--file Dockerfile.docs .
```

```
docker container run \
--rm \
-v $PWD:/code \
-w /code \
mkdocs:latest \
mkdocs build
```
The documentation will be created under the `/docs` folder, you can open the `index.html` file in a browser and navigate the generated document there.

Signed-off-by: Emilio Reyes <emilio.reyes@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/edgexfoundry/ci-management/blob/master/.github/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Added labels
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [X] Documentation content changes
- [ ] Other... Please describe:

## Issue Number:
fix 319

## Sandbox Testing
Not applicable
Test Links :

### CODE REVIEW

**Review by cjoyv — APPROVED** [review]
LGTM

File: vars/edgeXBuildGoApp.groovy [hunk 1]
```diff
@@ -17,14 +17,44 @@ import org.jenkinsci.plugins.workflow.libs.Library
...
+ dockerBuildContext | optional | str | The path for Docker to use as its build context when building your project. This applies to building both the CI build image and project image.<br /><br />**Default**: `.`
+ dockerBuildArgs | optional | list | The list of additonal arguments to pass to Docker when building the image for your project.<br /><br />**Default**: `[]`
+ dockerNamespace | optional | str | The docker registry namespace to use when publishing Docker images. **Note** for edgeX projects this value should be empty string.<br /><br />**Default**: `''`
+ dockerImageName | optional | str | The name of the Docker image for your project.<br /><br />**Default**: `docker-${project}`
+ dockerNexusRepo | optional | str | The name of the Docker Nexus repository where the project Docker image ${dockerImageName} will be published to if ${pushImage} is set.<br /><br />**Default**: `staging`
+ buildImage | optional | bool | Specify if Jenkins should build a Docker image for your project.<br /><br />**Default**: `true`
+ pushImage | optional | bool | Specify if Jenkins should push your project's image to ${dockerNexusRepo}.<br /><br />**Default**: `true`
```
Comment (ernestojeda) [inline vars/edgeXBuildGoApp.groovy:48]: I would replace ${dockerNexusRepo} with `dockerNexusRepo`. Also note, if `buildImage` is false `pushImage` will also be set to false.

File: vars/edgeXBuildGoApp.groovy [hunk 2]
```diff
@@ -17,14 +17,44 @@ import org.jenkinsci.plugins.workflow.libs.Library
...
+ swaggerApiFolders | optional | list | The list of paths to your projects API Swagger-based documentation.<br /><br />**Default**: `['openapi/v1']`
+ failureNotify | optional | str | The group emails (comma-delimited) to email when the Jenkins job fails.<br /><br />**Default**: `edgex-tsc-core@lists.edgexfoundry.org,edgex-tsc-devops@lists.edgexfoundry.org`
+ buildExperimentalDockerImage | optional | bool | Specify if Jenkins should add an additonal GitHub tag called `experimental` at the same commit where the semantic version is tagged. **Note** this feature is currently only used internally for DevOps builds.<br /><br />**Default**: `false`
+ artifactTypes | optional | list | A list of types that the Jenkins build will designate as artifacts, valid list values are `docker` and `archive`. **Note** if `archive` is specified then all `tar.gz` or `zip` files that your project build creates in the ${artifactRoot} folder will be archived to Nexus.<br /><br />**Default**: `['docker']`
```
Comment (ernestojeda) [inline vars/edgeXBuildGoApp.groovy:56]: I would replace ${artifactRoot} with `artifactRoot`

File: vars/edgeXBuildGoApp.groovy [hunk 3]
```diff
@@ -17,14 +17,44 @@ import org.jenkinsci.plugins.workflow.libs.Library
...
+ dockerFilePath | optional | str | The path to the Dockerfile for your project.<br /><br />**Default**: `Dockerfile`
+ dockerBuildFilePath | optional | str | The path to the Dockerfile that will serve as the CI build image for your project.<br /><br />**Default**: `Dockerfile.build`
+ dockerBuildContext | optional | str | The path for Docker to use as its build context when building your project. This applies to building both the CI build image and project image.<br /><br />**Default**: `.`
+ dockerBuildArgs | optional | list | The list of additonal arguments to pass to Docker when building the image for your project.<br /><br />**Default**: `[]`
+ dockerNamespace | optional | str | The docker registry namespace to use when publishing Docker images. **Note** for edgeX projects this value should be empty string.<br /><br />**Default**: `''`
```
Comment (ernestojeda) [inline vars/edgeXBuildGoApp.groovy:44]: I think the note should read: Note for EdgeX projects images are published to the root of the docker registry and thus the namespace should be empty.

File: vars/edgeXBuildGoParallel.groovy [hunk 4]
```diff
@@ -17,15 +17,41 @@ import org.jenkinsci.plugins.workflow.libs.Library
...
+ dockerFileGlobPath | optional | str | The pattern for finding Dockerfiles to build. **Note** Docker images will be named with the same name as the directory which the Dockerfile was found in with a `docker-` prefix and `-go` suffix. Example: `docker-<folder>-go`<br /><br />**Default**: `cmd/** /Dockerfile`
+ dockerImageNamePrefix | optional | str | The prefix to apply to the names of all the Docker images built.<br /><br />**Default**: `docker-`
+ dockerImageNameSuffix | optional | str | The suffix to apply to the names of all the Docker images built.<br /><br />**Default**: `-go`
+ dockerBuildFilePath | optional | str | The path to the Dockerfile that will serve as the CI build image for your project.<br /><br />**Default**: `Dockerfile.build`
+ dockerBuildContext | optional | str | The path for Docker to use as its build context when building your project. This applies to building both the CI build image and project image.<br /><br />**Default**: `.`
+ dockerNamespace | optional | str | The docker registry namespace to use when publishing Docker images. **Note** for edgeX projects this value should be empty string.<br /><br />**Default**: `''`
```
Comment (ernestojeda) [inline vars/edgeXBuildGoParallel.groovy:45]: Same note as edgeXBuildGoApp

File: vars/edgeXBuildGoParallel.groovy [hunk 5]
```diff
@@ -17,15 +17,41 @@ import org.jenkinsci.plugins.workflow.libs.Library
...
+ dockerNexusRepo | optional | str | The name of the Docker Nexus repository where the project Docker image ${dockerImageName} will be published to if ${pushImage} is set.<br /><br />**Default**: `staging`
```
Comment (ernestojeda) [inline vars/edgeXBuildGoParallel.groovy:46]: Same note as edgeXBuildGoApp

File: vars/edgeXBuildGoParallel.groovy [hunk 6]
```diff
@@ -17,15 +17,41 @@ import org.jenkinsci.plugins.workflow.libs.Library
...
+ buildImage | optional | bool | Specify if Jenkins should build a Docker image for your project.<br /><br />**Default**: `true`
+ pushImage | optional | bool | Specify if Jenkins should push your project's image to ${dockerNexusRepo}.<br /><br />**Default**: `true`
```
Comment (ernestojeda) [inline vars/edgeXBuildGoParallel.groovy:48]: Same note as edgeXBuildGoApp

**Review by ernestojeda — COMMENTED** [review]
See my comments.

**Review by ernestojeda — APPROVED** [review]
LGTM

**Pull request merged**